ENIGMAsystem/SHELL/Graphics_Systems/Direct3D11/DX11screen.cpp

```diff
@@ -46,6 +36,14 @@ void scene_end() {
 
 }
 
+unsigned char* graphics_copy_screen_pixels(int x, int y, int width, int height, bool* flipped) {
+  return nullptr;
```

Comment:
Return a width×height black texture if it's actually prohibitive to implement this. In fact, why not just have the caller specify the buffer to allow arena allocation / dense storage / whatever.

Reply:
I can do that so that D3D11 at least doesn't crash when those functions are called. Also, we could do that but I'd want to change the other texture interface too, so I'll wait and would rather do that in one go. However, it's honestly better to do #1721 DMA pixel transfer instead so that it speeds up everything. That now includes OpenGL and Direct3D since we would inform the caller of the row pitch, and GL wouldn't have to perform the additional memory copy to return a contiguous subregion of pixels.

Reply:
Ok: 7cef5d699fcb943ea69d6bb6c78b826489092c15

I made it do a dummy allocation for now. I actually can't see what's going on in D3D11 due to a MinGW shader compiler regression I am waiting to be fixed, but I can tell you it doesn't crash now. Later D3D11 will need reorganized like D3D9 was here to consolidate the screen/render target/texture transferring logic.
https://github.com/msys2/MINGW-packages/issues/5330